src/hooks/useOrds.ts

```diff
@@ -598,6 +611,157 @@ export const useOrds = () => {
+
+      // output 0
+      const buyerOutput = new TxOut(BigInt(1), P2PKHAddress.from_string(ordAddress).get_locking_script());
+      purchaseTx.add_output(buyerOutput);
+
+      // output 1
+      const payOutput = TxOut.from_hex(Buffer.from(payout, 'base64').toString('hex'));
+      purchaseTx.add_output(payOutput);
+
+      // output 2 - change
+      const dummyChangeOutput = new TxOut(
+        BigInt(0),
+        P2PKHAddress.from_string(fundingAndChangeAddress).get_locking_script(),
+      );
+      purchaseTx.add_output(dummyChangeOutput);
+
+      // output 3 - marketFee
+      const dummyMarketFeeOutput = new TxOut(
+        BigInt(0),
+        P2PKHAddress.from_string(marketplaceAddress).get_locking_script(),
+      );
+      purchaseTx.add_output(dummyMarketFeeOutput);
+
+      let preimage = purchaseTx.sighash_preimage(
+        SigHash.InputOutput,
+        0,
+        Script.from_bytes(Buffer.from(script, 'base64')),
+        BigInt(1), //TODO: use amount from listing
+      );
+
+      listingInput.set_unlocking_script(
+        Script.from_asm_string(
+          `${purchaseTx.get_output(0)!.to_hex()} ${purchaseTx.get_output(2)!.to_hex()}${purchaseTx
+            .get_output(3)!
+            .to_hex()} ${Buffer.from(preimage).toString('hex')} OP_0`,
+        ),
+      );
+      purchaseTx.set_input(0, listingInput);
+
+      let marketFee = Math.ceil(price * marketplaceRate);
+      let fee = calculateFee(1, purchaseTx);
+      let satsNeeded = fee + price + marketFee;
+      const inputs = getInputs(fundingUtxos, satsNeeded, false);
+      const satsCollected = inputs.reduce((a, r) => a + r.satoshis, 0);
+      const changeAmt = Math.ceil(satsCollected - satsNeeded);
+
+      const changeOutput = new TxOut(
+        BigInt(changeAmt),
+        P2PKHAddress.from_string(fundingAndChangeAddress).get_locking_script(),
+      );
+
+      purchaseTx.set_output(2, changeOutput);
+
+      // add output 3 - market fee
+      const marketFeeOutput = new TxOut(
+        BigInt(marketFee),
+        P2PKHAddress.from_string(marketplaceAddress).get_locking_script(),
+      );
+      purchaseTx.set_output(3, marketFeeOutput);
+
+      preimage = purchaseTx.sighash_preimage(
+        SigHash.InputOutputs,
+        0,
+        Script.from_bytes(Buffer.from(script, 'base64')),
+        BigInt(1),
+      );
+
+      listingInput.set_unlocking_script(
+        Script.from_asm_string(
+          `${purchaseTx.get_output(0)!.to_hex()} ${purchaseTx.get_output(2)!.to_hex()}${purchaseTx
+            .get_output(3)!
+            .to_hex()} ${Buffer.from(preimage).toString('hex')} OP_0`,
+        ),
+      );
+      purchaseTx.set_input(0, listingInput);
+
+      inputs.forEach((utxo, idx) => {
+        const fundingInput = new TxIn(Buffer.from(utxo.txid, 'hex'), utxo.vout, Script.from_asm_string(utxo.script));
+        purchaseTx.add_input(fundingInput);
+
+        const sig = purchaseTx.sign(
+          payPk,
+          SigHash.InputsOutputs,
```

Comment:
Signing this as `InputsOutputs` means the second pass of the loop will add an input which will invalidate the sig from the previous pass through the loop. `InputOutpus` will resolve that